app/models/order.rb

```diff
@@ -67,5 +67,14 @@ def update_notification_checked!(notification)
     notification.update({ checked: true }) if notification.where(checked: false)
   end
 
-
+  def save_notification_checked!(current_user)
+    chats = Order.find(self.id).chats
+    chats.each do |chat|
+      # notification = Notification.where(visitor_id: current_user.id, chat_id: chat.id, action: "commented")
+      notification = Notification.where("visitor_id = ? and chat_id = ? and action = ? ", current_user.id, chat.id, "commented")
+      if notification.visitor_id == notification.visited_id
+        notification.checked = true
+      end
```

Comment:
こんな感じかと。後、メソッド名にsaveってついてるので、saveした方がいいのではと思います。ここら辺は要件がわからないので必要可否はお任せします。

```suggestion
      notifications = Notification.where("visitor_id = ? and chat_id = ? and action = ? ", current_user.id, chat.id, "commented")
      notifications.each do |notification|
        if notification.visitor_id == notification.visited_id
          notification.checked = true
          # saveが必要では？
          notification.save!
        end
      end
```